qiskit_optimization/algorithms/grover_optimizer.py

```diff
@@ -199,7 +237,7 @@ def solve(self, problem: QuadraticProgram) -> OptimizationResult:
         # Initialize oracle helper object.
         qr_key_value = QuantumRegister(self._num_key_qubits + self._num_value_qubits)
         orig_constant = problem_.objective.constant
-        measurement = not self.quantum_instance.is_statevector
+        measurement = not (self._quantum_instance and self._quantum_instance.is_statevector)
```

Comment:
`self._quantum_instance` this should ideally be an explicit test against None since that's the intent here I believe.

Reply:
@a-matsuo What do you think of this?

Reply:
In the beginning of `solve()`, we have the following code to ensure only one of a sampler or quantum instance is given. Do you mean that still it's better to explicitly write like `self._sampler and not self._quanutm_instance`?
```
      if self._sampler is None and self._quantum_instance is None:
          raise ValueError("A quantum instance or sampler must be provided.")

      if self._quantum_instance is not None and self._sampler is not None:
          raise ValueError("Only one of quantum_instance or sampler can be passed, not both!")
```

Reply:
This is just about the part `not(self._quantum_instance..) above and below (if self._sampler) - the recommendation is when comparing against None to do an explicit test https://peps.python.org/pep-0008/#programming-recommendations. Its being done in other tests as I would expect - just not these places I noted. I don't mean change the logic around as such just make the test itself, which is doing it based on None or None - explicit rather than the bool conversion.

Reply:
Do you mean the following?
```suggestion
        measurement = self._quantum_instance is None or not self._quantum_instance.is_statevector
```

Reply:
Yes that works